packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -517,6 +528,16 @@ export class PolicyStatement extends cdk.Token {
       return result;
     }
   }
+
+  /**
+   * The hash of this statement. Used to avoid duplicates.
+   */
+  public get hash(): string {
```

Comment:
I like the idea of having a `hash` for a statement, but this is actually not necessary for comparison purposes (`JSON.stringify(this.toJson())` is sufficient). Opinions?

Reply:
This is IAM, which is sensitive, so I am definitely not comfortable with the idea of making a hash-comparison, because in the improbable cause of a collision, it'll have awkward consequences.

I am however not sure that `JSON.stringify` will work though, as there can be any amount of unresolved `Token`s here that may or may not be represented by the same `string`.

Reply:
> I am however not sure that `JSON.stringify` will work though, as there can be any amount of unresolved `Token`s here that may or may not be represented by the same `string`.

If this is the case then the comparison will fail and it will leave the statements as is, which is suboptimal but safe?

Reply:
IIRC, `toJson` will throw in this case...